src/modm/math/proportional_unsigned.hpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2022, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <algorithm>
+#include <modm/math/utils/integer_traits.hpp>
+
+#ifdef PROPORTIONAL_UNSIGNED_DEBUG
+#include <modm/architecture/interface/assert.hpp>
+
+static modm::Abandonment
+bad_gray_value(const modm::AssertionInfo &info)
+{
+	if(!strncmp(info.name, (char*)"propun", 6)) {
+    	MODM_LOG_ERROR << info.name << " " << info.description << " " << info.context << modm::endl;;
+    	return modm::Abandonment::Ignore;
+	}
+    return modm::Abandonment::DontCare;
+}
+
+// Register handler for all builds
+MODM_ASSERTION_HANDLER(bad_gray_value);
+// Or register only for debug builds
+// MODM_ASSERTION_HANDLER_DEBUG(log_assertion);
+#endif
+
```

Comment:
I would use a `modm_assert_continue_ignore_debug` that only triggers an assertion in debug mode and otherwise continues.
I would provide this handler code with `MODM_ASSERTION_HANDLER_DEBUG()` in the documentation instead.
Otherwise if you provide the handler here by default, you cannot disable it.

```cpp
if(not modm_assert_continue_ignore_debug(value <= max, "propun.ctor", "value exceeded max", max)) {
    value = max;
}
```

I don't think it's necessary for the release build or to provide a default debug assertion, since you will likely very much see the issue graphically anyways, which will probably cause people to check their input, and then maybe check the documentation and find the debug handler example code.

Reply:
Just be careful about the constexpr, since the `modm_assert` mechnism is obviously not `constexpr` (would be interesting if that can be redirected to a `static_assert` with `std::is_constant_evaluated`, but not sure)!

Reply:
Also put the handler code into the example of course.

Reply:
Oh, I see. You can guard this code with `#ifdef MODM_DEBUG_BUILD ` which is set in debug profile. (which apparently is undocumented?!?)

Reply:
> I would provide this handler code with MODM_ASSERTION_HANDLER_DEBUG() in the documentation instead
Moved the handler to `proportional_unsigned.md`

Reply:
> Oh, I see. You can guard this code with `#ifdef MODM_DEBUG_BUILD ` which is set in debug profile. (which apparently is undocumented?!?)

👉 Indeed! This is what i've been looking for. And is not documented.. at least not at https://modm.io/reference/module/modm-architecture-assert/ where it belongs to by any means

Reply:
Yup, I'm already fixing it and adding it to `modm/architecture/utils.hpp` and `:build` module too.